com.unity.netcode.gameobjects/Prototyping/NetworkTransform.cs

```diff
@@ -398,17 +446,12 @@ private void FixedUpdate()
                 return;
             }
 
-            if (CanUpdateTransform && IsNetworkStateDirty(ReplNetworkState.Value))
+            if (CanUpdateTransform)
             {
-                UpdateNetworkState();
+                ReplNetworkState.SetDirty(UpdateNetworkState(ReplNetworkState.Value));
             }
-            else
+            else if (UpdateNetworkState(PrevNetworkState))
```

Comment:
mmm not a big fan of this
The word update here implies a state change somewhere, while this if check should only do a check. If !CanUpdateTransform, we should never "update the transform".

Reply:
`CanUpdateTransform` will be deleted entirely soon and we'll do server-driven only. Let's hope this way is temporary.

Reply:
more specifically PR #1059